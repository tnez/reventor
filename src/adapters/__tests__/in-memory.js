const { InMemory } = require('..')

const factory = require('../__factories__')

const events = [
  factory.event({
    aggregateId: 'account:1234',
    aggregateVersion: 0,
    correlationId: 'a-uuid-1',
    eventType: 'account:created',
  }),
  factory.event({
    aggregateId: 'account:1234',
    aggregateVersion: 1,
    correlationId: 'a-uuid-2',
    data: {
      amount: 100,
    },
    eventType: 'account:deposit',
  }),
  factory.event({
    aggregateId: 'account:9999',
    aggregateVersion: 0,
    correlationId: 'a-uuid-3',
    eventType: 'account:created',
  }),
]

describe('read', () => {
  const store = new InMemory(events)

  it('returns events for aggregate id', async () => {
    const result = await store.read('account:1234')
    expect(result).toMatchSnapshot()
  })
})

describe('write', () => {
  const store = new InMemory()

  it('resolves to undefined', async () => {
    await expect(store.write(events)).resolves.toBeUndefined()
  })
})
