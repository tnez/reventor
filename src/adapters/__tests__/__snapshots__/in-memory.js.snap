// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`read returns events for aggregate id 1`] = `
Array [
  Object {
    "aggregateId": "account:1234",
    "aggregateVersion": 0,
    "correlationId": "a-uuid-1",
    "data": Object {},
    "eventType": "account:created",
  },
  Object {
    "aggregateId": "account:1234",
    "aggregateVersion": 1,
    "correlationId": "a-uuid-2",
    "data": Object {
      "amount": 100,
    },
    "eventType": "account:deposit",
  },
]
`;
